test/ConfigAggregatorTest.php

```diff
@@ -90,10 +90,69 @@ function () {
                 return ['foo' => 'bar', ConfigAggregator::ENABLE_CACHE => true];
             }
         ], $this->cacheFile);
-        $this->assertTrue(file_exists($this->cacheFile));
+        $this->assertFileExists($this->cacheFile);
         $cachedConfig = include $this->cacheFile;
-        $this->assertInternalType('array', $cachedConfig);
-        $this->assertEquals(['foo' => 'bar', ConfigAggregator::ENABLE_CACHE => true], $cachedConfig);
+
+        $this->assertIsArray($cachedConfig);
+        $this->assertSame(['foo' => 'bar', ConfigAggregator::ENABLE_CACHE => true], $cachedConfig);
+    }
+
+    public function testConfigAggregatorCanCacheConfigWithClosures()
+    {
+        new ConfigAggregator([
+                function () {
+                    return [
+                     'toUpper' => function ($input) {
+                        return strtoupper($input);
+                     },
+                     ConfigAggregator::ENABLE_CACHE => true
+                    ];
+                }
+         ], $this->cacheFile);
+
+        $this->assertFileExists($this->cacheFile);
+
+        $cachedConfig = include $this->cacheFile;
+
+        $this->assertIsCallable($cachedConfig['toUpper']);
+        $this->assertSame('FOOBAR', call_user_func($cachedConfig['toUpper'], 'foobar'));
```

Comment:
`call_user_func()` usage should can be changed with direct call: 

```suggestion
        $this->assertSame('FOOBAR', $cachedConfig['toUpper']('foobar'));
```

ref https://3v4l.org/49DtW

Reply:
Fixed in my latest commit.